cmake/compilerFlags.cmake

```diff
@@ -33,7 +33,9 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
     if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
         # This fails under Fedora - MinGW - Gcc 8.3
         if (NOT (MINGW OR CYGWIN OR CMAKE_HOST_SOLARIS))
+            if (NOT APPLE) # Don't know why this isn't working correctly on Apple with M1 processor

```

Comment:
@clanmills is this maybe worth tracking in an issue?  
Also didn't this now disable `-fstack-clash-protection` for all Apple machine? e.g. also for the normal x86 platforms?